jaraco/collections/__init__.py

```diff
@@ -119,7 +126,7 @@ def dict_map(function, dictionary):
     return dict((key, function(value)) for key, value in dictionary.items())
 
 
-class RangeMap(dict):
+class RangeMap(Dict[int, _VT]):
```

Comment:
Oh. I didn't know you could subclass from `Dict`. TIL.

`int` is overspecified here. RangeMap accepts any keys (possibly heterogeneous) that are comparable based on the key_match_comparator.

Also, the values don't need to be uniform. They can be heterogeneous.

Reply:
> Also, the values don't need to be uniform. They can be heterogeneous.

Are RangeMap meant to be mutable? If not I think it would make sense to make the value covariant like a `Mapping`

Reply:
> Is `RangeMap` meant to be mutable?

I was not intending for it to be restricted to be immutable. I can't recall if I've used it for an application that changes it dynamically, but it's a use-case that should be supported unless there's a practical benefit to making it immutable. It was modeled after the `dict`, so is meant to have similar flexibility.

Reply:
Alright, in that case I *think* it's correct to key both keys and values invariant.